lte/gateway/python/magma/mobilityd/ip_address_man.py

```diff
@@ -134,31 +134,35 @@ def __init__(self,
         if not persist_to_redis:
             self._assigned_ip_blocks = set()  # {ip_block}
             self.sid_ips_map = defaultdict(IPDesc)  # {SID=>IPDesc}
+            self._dhcp_gw_info = UplinkGatewayInfo(defaultdict(str))
+
         else:
             if not redis_port:
                 raise ValueError(
                     'Must specify a redis_port in mobilityd config.')
             client = get_default_client()
             self._assigned_ip_blocks = store.AssignedIpBlocksSet(client)
             self.sid_ips_map = store.IPDescDict(client)
+            self._dhcp_gw_info = UplinkGatewayInfo(store.GatewayInfoMap())
 
         self.ip_state_map = IpDescriptorMap(persist_to_redis, redis_port)
         logging.info("Using allocator: %s", self.allocator_type)
+
         if self.allocator_type == MobilityD.IP_POOL:
+            self._dhcp_gw_info.read_default_gw()
             self.ip_allocator = IpAllocatorStatic(self._assigned_ip_blocks,
                                                   self.ip_state_map,
                                                   self.sid_ips_map)
         elif self.allocator_type == MobilityD.DHCP:
             dhcp_store = store.MacToIP()  # mac => DHCP_State
```

Comment:
does `dhcp_store` only applies for stateless mobilityd? should we also add it for stateful mode as a in memory dict?

Reply:
not really, it is same for both state and stateless. it is there at line 137

Reply:
without persistent mapping dhcp_store is not that useful, since we will loose DHCP address mapping that we got for all UEs.

Reply:
Makes sense, it would be good to specify it on the docs that there's a dependency for DHCP allocation mode on mobilityd having persistence on db as our default operation is still using in-memory storage.